pilot/pkg/model/sidecar.go

```diff
@@ -604,11 +604,11 @@ func (ilw *IstioEgressListenerWrapper) selectVirtualServices(virtualServices []c
 				// Check if the hostnames match per usual hostname matching rules
 				hostFound := false
 				for _, h := range rule.Hosts {
-					// TODO: This is a bug. VirtualServices can have many hosts
-					// while the user might be importing only a single host
-					// We need to generate a new VirtualService with just the matched host
+					// VirtualServices can have many hosts while the user might be importing only a single host
+					// so we need to generate a new VirtualService with just the matched host
 					if importedHost.Matches(host.Name(h)) {
-						importedVirtualServices = append(importedVirtualServices, c)
+						newVSConfig := newVSConfigHostsSetting(c, h)
```

Comment:
Concerned about the performance here, prefer if all hosts match, not deepcopy

Reply:
yes, mentioned above, there will be a new submit. Thanks for your reply. ^_^